### ISSUE TITLE
Support for server side pre-rendering

### ISSUE THREAD

**Issue by jmezach** [urish/angular2-moment#82]
I've been working on an Angular 2 app with an ASP.NET Core backend that uses server side pre-rendering. Unfortunately I found out that using these pipes breaks the server side rendering.

This mostly seems to be because they use window.setTimeout and window.clearTimeout, but window isn't available during the server side rendering for obvious reasons. Is there something we can do here to make this play nice with server side pre-rendering? Perhaps checking if window is defined would be enough?

**Comment by urish** [comment]
This makes sense. Can you please send a PR for that?

Also, please see #74

**Comment by jmezach** [comment]
I started working on this, but I'm having a hard time trying it out. I've tried to npm link it, but then the server side pre-rendering fails entirely with the same error message that is mentioned in #68. Any thoughts on that?

**Comment by urish** [comment]
How did you import the module?

Please see my last comment on #81

**Comment by jmezach** [comment]
I've tried both:

```
import { MomentModule } from 'angular2-moment/module';
```

and:

```
import { MomentModule } from 'angular2-moment/module';
```

But I'm getting the same error in both cases?

**Comment by urish** [comment]
Can you please check with the latest beta, `1.0.0-beta.3`?

**Comment by jmezach** [comment]
I merged my forks master with your master and then tried to npm link again, but I'm still getting the same error message.

I noticed that both the TimeAgoPip and CalendarPipe now use ngZone.runOutsideAngular(), but it looks like that is still being executed during server side pre-rendering so it doesn't fix this issue. I guess we'll have to check for the existence of window explicitly. Want to try that out locally first though.
